File: CareConnect/CareConnect.Models/Responses/KpiData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Responses
{
    public class KpiData
    {
        public int TotalNewClients { get; set; }

        public int TotalAppointments { get; set; }
        
        public int TotalWorkshops { get; set; }
        
        public double NewClientsChange { get; set; }
        
        public double AppointmentsChange { get; set; }
        
        public double WorkshopsChange { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/responses/kpi_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'kpi_data.g.dart';

@JsonSerializable()
class KpiData {
  final int totalNewClients;
  final int totalAppointments;
  final int totalWorkshops;
  final double newClientsChange;
  final double appointmentsChange;
  final double workshopsChange;

  KpiData({
    required this.totalNewClients,
    required this.totalAppointments,
    required this.totalWorkshops,
    required this.newClientsChange,
    required this.appointmentsChange,
    required this.workshopsChange,
  });

  factory KpiData.fromJson(Map<String, dynamic> json) =>
      _$KpiDataFromJson(json);

  Map<String, dynamic> toJson() => _$KpiDataToJson(this);
}
